RocketChat/Rocket.Chat
# [NEW] Option to prevent users from using Invisible status

## Proposed changes (including videos or screenshots)


Add an `admin` option to allow/disallow the `Invisible` status option from all users. This option is available in the `Accounts` section.

![2021-01-06-11-55-22](https://user-images.githubusercontent.com/49413772/103782988-ebc52300-5016-11eb-8a29-dd540c21e11c.gif)

If the option is turned off, the `users.setStatus` endpoint is also restricted from users trying to change their status to `Invisible`, throwing the following error:
```json
{
    "success": false,
    "error": "Invisible status is disabled [error-not-allowed]",
    "stack": "Error: Invisible status is disabled [error-not-allowed]\n    at DDPCommon.MethodInvocation.<anonymous> (app/api/server/v1/users.js:425:13)\n    at packages/dispatch_run-as-user.js:211:14\n    at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)\n    at Object.Meteor.runAsUser (packages/dispatch_run-as-user.js:210:33)\n    at Object.post (app/api/server/v1/users.js:415:10)\n    at app/api/server/api.js:394:82\n    at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1234:12)\n    at Object._internalRouteActionHandler [as action] (app/api/server/api.js:394:39)\n    at Route.share.Route.Route._callEndpoint (packages/nimble_restivus/lib/route.coffee:150:32)\n    at packages/nimble_restivus/lib/route.coffee:59:33\n    at packages/simple_json-routes.js:98:9",
    "errorType": "error-not-allowed",
    "details": {
        "method": "users.setStatus"
    }
}
```


## Issue(s)

Fixes https://github.com/RocketChat/Rocket.Chat/issues/5020

## Steps to test or reproduce


1.  Login as a user that has the `admin` role
2. Go to the `Administration`->`Accounts` section
3. Disable the option to see the changes in the user's status


## Further comments
